introduction_and_learning/Histogram/histogram_api.py

```diff
@@ -0,0 +1,90 @@
+__all__ = [
+    "HistogramInterface"
+]
+from abc import abstractmethod, ABCMeta
+
+
+class HistogramInterface(metaclass=ABCMeta):
+    """
+    HistogramInterface is basic interface for creating histogram from various
+    file types classes.
+
+    HistogramInterface is abstract class. Methods that required are
+        read_data(file_path)
+        create(data_input)
+        generate_report(histogram, output_type, path)
+    """
+
+    @abstractmethod
+    def read_data(self, file_path: str) -> str:
+        """
+        This method handling opening and reading from given file. Type of this
+        file is given in class definition.
+
+        Parameters
```

Comment:
I am not sure if we need to describe parameters etc. here, most of the time, abstract methods should be described shortly. This is not an interface toward the user, but rather towards programmer. Probably shorter description would be better, and longer one in public classes that user will be using.